### DISCUSSION TITLE
Peanut pod-filling and fractional available soil water calculation

### DISCUSSION THREAD

**Discussion by ParameshwariS-tiger** [APSIMInitiative/ApsimX#8826] [Q&A] [answered]
Hi all, for my study I am required to calculate 0.6 pod-filling stage of peanuts that is, last 4-6 weeks before harvest. And from this stage I need to calculate the days of water stress that is less than 0.2 fractional available soil water.

For this I have used the Peanut module which provides reporting variables including markers for various stages of peanut development and soil water content. I have particularly used the below reporting variables which was useful for my calculation: 

- _[Peanut].Phenology.CurrentStageName_
- _[Soil].SoilWater.SW_

Using this output I have made calculations in the following way:

- First, I computed the duration between the 'StartPodDevelopment' and 'HarvestRipe' stages from _Phenology.CurrentStageName_.
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/0c186cd9-37dc-42f0-b2b5-d3316fa70ef1)

- Next, I multiplied this duration by 0.6 to obtain the number of days corresponding to the last 0.6 pod-filling stage.
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/079f0da7-95db-411e-9c07-da4adf3fd73c)
                                                                                         .
                                                                                         .
                                                                                         .
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/53206ea5-7e5c-4aca-a9a8-86b4eac5e7f2)

- Finally, I filtered the days where the _SoilWater.SW_ was less than 0.2
![image](https://github.com/APSIMInitiative/ApsimX/assets/147519876/29ed7e03-3fa4-45a8-a018-dd6a50307b24)

Does this approach sound right? It would be great if someone could help me regarding this, incase my approach is incorrect or I am misinterpreting some aspects in terms of my understanding. Thanks:)

**Comment by hol353** [comment]
> Finally, I filtered the days where the SoilWater.SW was less than 0.2

SoilWater is a volumetric water content i.e. the amount of water per unit volume of soil. A value of 0.2 may or may not be water stressed. It will depend on how the soil LL15 and DUL are defined. It is probably better to get APSIM to output:

```
sum([PeanutSoil].PAWmm)
sum([PeanutSoil].PAWCmm)
```

PAWmm is the amount of water (mm) available to the peanut plant. PAWC is the maximum amount of water (mm) available water to the peanut plant. The sum is necessary because they are a soil profile layered variable (i.e. one value per layer). You could then calculate a stress factor:

```c#
sum([PeanutSoil].PAWmm) / sum([PeanutSoil].PAWCmm)
```
A value of 0 means there is no available water. A value of 1 means the profile is full. The APSIM report model can calculate this variable for you by typing the above into REPORT.

APSIM can also output water stress as defined by the leaf model. The variable is [Peanut].Leaf.FW. This is a 0 to 1 variable where 1 is no water stress and 0 is maximum water stress. 

In the source code FW is define as:
```c#
FW = Water allocated to leaf / PotentialEP
```

Whether you use FW or the PAW approach depends on how you want to define water stress. 

> I filtered the days where the SoilWater.SW was less than 0.2

I assume you mean the filter is something like:

```
Clock.Today >= StartPodDevelopment AND
Clock.Today <= StartPodDevelopment + 0.6 * (EndPodDevelopment - StartPodDevelopment) AND
WaterStress < 0.2
```

**Reply by ParameshwariS-tiger** [reply]
@hol353 Thanks for the detailed clarification on water stress
I have exported the output from APSIM and did all the filters and calculation in python pandas; calculated 0.6 pod filling stage and then filtered it by taking df[Soil.SoilWater.SW(1)] < 0.2

**Reply by hol353** [reply]
Soil.SoilWater.SW(1) doesn't seem right to me. SW is a volumetric water content, not a fraction 0 to 1. Also SW(1) will be the 2nd layer of the soil. Is this what you want?

**Reply by ParameshwariS-tiger** [reply]
I want to filter the days where fractional available soil water is less than 0.2, and find soil temperature for those at the depth of 5cm

**Comment by hol353** [comment]
Use [PeanutSoil].PAW for available soil water (not SW)

**Reply by ParameshwariS-tiger** [reply]
@hol353 So can I use `sum([PeanutSoil].PAW)` instead of `sum([PeanutSoil].PAWmm) / sum([PeanutSoil].PAWCmm)`  for fractional available soil water ? I actually tried both but I am not getting between the range 0 to 1, some values exceed 1

**Comment by hol353** [comment, marked as answer]
No, ```sum([PeanutSoil].PAW)``` returns an amount of available water (volumetric) - not what you want. Use ```sum([PeanutSoil].PAWmm) / sum([PeanutSoil].PAWCmm)''' to get a fraction of water. Yes it will go briefly above one when the soil is saturated but it should drain quite quickly.

**Reply by ParameshwariS-tiger** [reply]
Okay got it, thanks a lot @hol353 for clarifying all my doubts

**Comment by ParameshwariS-tiger** [comment]
@hol353 For APSIM results such as soil temperature and soil water, are the values averaged over the day, or are the daily minimum and maximum values provided?

If it is averaged values, is it possible to get minimum or maximum values over the day?

**Comment by hol353** [comment]
SoilTemperature has SoilTemperature.Value, SoilTemperature.MinSoilTemp and SoilTemperature.MaxSoilTemp which are the daily average, minimum and maximum temps respectively.

The soil water model (like most models in APSIM) only calculates a single soil water value per day which you can think of as an average. There are no minimum or maximum valuese calculated.

**Reply by ParameshwariS-tiger** [reply]
Got it thanks, what are the report variables exactly? Because I am using `[Soil].Temperature.Value` currently which is working but `[Soil].Temperature.MinSoilTemp` and` [Soil].Temperature.MaxSoilTemp`  are invalid

And where should we find the list of available report variables, because in the user interface I was not able to find minimum and maximum values for soil temperature?

**Comment by hol353** [comment]
It sounds like you're using the older soil temperature model - CERESSoilTemperatue. If you hover the mouse over the 'Temperature' node under the soil you should see what type of model it is. Remove the old model and add 'SoilTemperature', then rename it to just 'Temperature' so that your reports will work as you've written them above.

**Reply by ParameshwariS-tiger** [reply]
Great it works now, thanks @hol353
